### ISSUE TITLE
Terminology issue

### ISSUE THREAD

**Issue by a-drenaline** [scipy/scipy#21357]
https://github.com/scipy/scipy/blob/87c46641a8b3b5b47b81de44c07b840468f7ebe7/scipy/signal/_signaltools.py#L2289

The function is called "hilbert"  -- one thinks this function only computes the Hilbert transform.
For example, the function "sum" -- one thinks the function output is the sum of its inputs.

But the "hilbert" does the Hilbert Transform and more. It computes the analytic signal. 

So I propose two options:
1. This function should be rename "analytic". 
2. The function needs further documentation to say, "hilbert" computes the analytic signal, in which the Hilbert transform is an intermediary step.

The wikipedia reference, which the function cites, clearly dictates the Hilbert transform being a part of the process in calculating the analytic signal.

![image](https://github.com/user-attachments/assets/81bc6c3b-c7d4-455e-bc4d-75648c7cffdb)

**Comment by lucascolley** [comment]
cc @DietBru @yagizolmez

**Comment by DietBru** [comment]
You are correct with your observation about the terminology, @a-drenaline. The function name is, as so many other inconsistencies, was established by [Matlab] and is used by many other signal processing libraries. IMO, having similar function names with the same functionality in different libraries is a great boon, which outweighs the naming inconsistencies in this case. Furthermore, renaming a function, which was added to SciPy [over 20 years ago], would probably break the code of many people.

> 2. The function needs further documentation to say, "hilbert" computes the analytic signal, in which the Hilbert transform is an intermediary step.

To be knit-picky: If you look at the source code, the hilbert transform is not calculated as an intermediary step, but is just a part of the resulting analytic signal. If you ignore the function name, the  [documentation] seems quite consistent to me in that regard.

Does this answer your questions, @a-drenaline? If so, feel free to close this bug.

[Matlab]: https://de.mathworks.com/help/signal/ref/hilbert.html
[documentation]: https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.hilbert.html
[over 20 years ago]: https://github.com/scipy/scipy/commit/009764630ecac95fd18aba30a1321cd5acdbc33c

**Comment by a-drenaline** [comment]
You are right to call out MATLAB. They do the same thing.
https://www.mathworks.com/help/signal/ref/hilbert.html

However, they are a company and own the code they write, thus they are likely to resist making changes.

This codebase, unlike MATLAB, has the opportunity to take feedback with higher likelihood of implementation.

I propose applying option 2 in this way:
Remove the Wikipedia article as the reference. Since as you mentioned, the Hilbert Transform is not computed in the code at all.

Add the documentation found in the MATLAB page, which explains their implementation of filtering out the negative frequencies, which is akin to the SciPy implementation.

![image](https://github.com/user-attachments/assets/5827e709-cbbe-4e11-932f-e7fc27505a2c)

**Comment by DietBru** [comment]
Would you care to do a pull request, @a-drenaline? 
Note that the formula in the [hilbert docstring] exactly formulates those steps from Matlab. Though there is no mentioning that it is the utilized calculation method.

[hilbert docstring]: https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.hilbert.html

**Comment by a-drenaline** [comment]
Agreed, the last part of your sentence does need to be in the update.

I don't see the pull request option when I click on the line in question.

In summary I proposed the following change
New Line 2289:
Compute the analytic signal, by filtering out the negative frequencies of
the input signal. See the Notes section for derivation.

**Comment by lucascolley** [comment]
@a-drenaline if you would like to make a pull request, we have quite extensive docs at https://scipy.github.io/devdocs/dev/contributor/development_workflow.html (and for basic things like submitting a pull request, the general GitHub docs are also very useful). For a simple change like this, the easiest set up may be to use [codespaces](https://docs.github.com/en/codespaces).

If you would prefer not to do that though, I'm sure someone else can submit the patch on your behalf.

**Comment by a-drenaline** [comment]
I tried pushing from an HTTPS clone.

It asks me for credentials. I use the credentials to login to Github.com, but it fails.

What are the right credentials?
![image](https://github.com/user-attachments/assets/2c60df44-824f-4863-ba06-d1f4f6fbdc33)

**Comment by nickodell** [comment]
You'll need to make a fork of this repo, then clone that fork and make your changes there. Once you've pushed to your fork, you can make a pull request.

**Comment by a-drenaline** [comment]
I did. 
I am at the last step where I need to push the change.

**Comment by nickodell** [comment]
Ah, I see. This can also be caused by having 2FA enabled. There are two workarounds, which you can read here.

https://stackoverflow.com/questions/17659206/git-push-results-in-authentication-failed

I personally use SSH based authentication, I find that works better.

**Comment by a-drenaline** [comment]
I tried the SSH approach, and it's not any better

![image](https://github.com/user-attachments/assets/dbda809b-4835-4f6d-ae08-d2fdec2a2299)

**Comment by dschmitz89** [comment]
You are trying to push to this repo. That is not possible. You need to use your own fork.

**Comment by lucascolley** [comment]
```shell
git remote rename origin upstream
git remote add origin {URL of your fork}
```

Then pushing to origin should work.

**Comment by ev-br** [comment]
Type `git remote -v` to see which remotes you have already. What you need is to push your branch to your scipy fork and send a pull request from the fork to the main scipy/scipy repository.

**Comment by lucascolley** [comment]
If `origin` and `upstream` are both pointing to the main `scipy` repo when you run `git remote -v`, you can fork the repo following https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo, `git remote rm origin`, and `git remote add origin {your fork}`.

In your post you wrote `git remote add origin {your branch}`. A branch is not the same as a fork, and the GitHub page linked in this comment should help explain that.
